### ISSUE TITLE
2 Phase Flow - New or updated data files

### ISSUE THREAD

**Issue by FLO-2DNoemi** [FLO-2DSoftware/qgis-flo-2d-plugin#869]
1.  - [x]  Changes to CON.DAT file

> MUD= 0 (no mudflow)
> MUD=1  (mudflow)
> MUD=2 (2 phase flow)

2. - [x]  For two phase flow both the sediment transport and mudflow variables have to be assigned in the SED.DAT file.  In other words, SED.DAT file should include Mudflow and Sediment lines, it can not be exclusive, see example below:

> M  0.0538 14.5 2.72 10.4 2.65 0.0
> C  8 0  0.576  3.71  2.65  14700.  0.0778  0  7362
> E  3.0

3. - [x]  INFLOW.DAT file reservoir line (R-line) needs additional data:

>The tailings material elevation or depth and the fluid elevation are assigned in INFLOW.DAT file in Line R as follows:

                                  Cell #             WS Elev.       Tailings Elev. or Depth    Reservoir Roughness            

                              R   7576                318.75            315.5                           0.20

4. - [x]  Individual cell tailings thickness can also be assigned in a new file (TAILINGS.DAT) which lists the reservoir grid element and the tailings thickness as shown below:

  7127      10.00   (= thickness below the cell bed elevation)
  7128      10.00
  7129      10.00
  7130      12.25
  7188      15.00
  7189      13.25
  7190      11.95
  7191      10.00
  7192      10.00

**Comment by FLO-2DJJ** [comment]
Items 1 and 2 included in functionality of plugin: reads/writes SED.DAT with 2 phase data. Mud/Sediment Transport dialog from FLO-2D toolbar includes Two Phase dynamics.

**Comment by FLO-2DJJ** [comment]
@FLO-2DNoemi , @FLO-2DKaren : 

1. Currently when exporting SED.DAT the file TAILINGS.DAT is not exported even if two phase flow is selected for sediment transport. TAILINGS.DAT is only exported when INFLOW.DAT is selected in the export components dialog AND the new 'Tailings Cells' layer has values. Is that correct? or we should change the logic of it? (By the way, the 'Tailings Cells' layer is filled using the new '2ph' tool in Grid Tools).
2. Similarly TAILINGS.DAT is only imported if it exists and 'inflows' is selected in the Import Components dialog. Is that correct?

In other words, importing/exported TAILINGS.DAT is linked to inflows and not sediment transport.

**Comment by FLO-2DNoemi** [comment]
@Juan Jose Rodriguez ***@***.***>

This is fine,  the tailings.dat file contains data for the tailing
material. The R line in the inflow.dat file turns on this.

**Comment by FLO-2DNoemi** [comment]
@FLO-2DJJ please review this as soon as possible:

1 and 2 are working.

3. INFLOW.DAT file reservoir line (R-line) is missing Tailings Elev. or Depth. This was already in the plugin, something was changed. Tailings.dat was also created before but now is not written.  This may be associated to INFLOW.DAT file R line issues, since if the R line (INFLOW.DAT) that turn on the writing code of the tailings.dat.

Prescribed failure data F and W lines are not created. This was working fine in previous plugins.

**Comment by FLO-2DJJ** [comment]
@FLO-2DNoemi , @FLO-2DKaren The plugin in GitHub writes R lines in INFLOW.DAT with tailings elev/depth. 
There is a new field in 'user_reservoirs' and 'reservoirs' tables (layers) with name 'tailings' (REAL DEFAULT -1.0). When a reservoir is created a dialog shows with default -1 (meaning 'no tailing value'):

![imagen](https://user-images.githubusercontent.com/20424575/190605413-9d9478ec-edd2-4613-8ea4-e8290bc90291.png)

A value different than -1 (for example 315.5) will make it to be written as:
R    7576    318.75    315.5    0.20

I believe the -1 default is equivalent to the 'use_n_value', saving one field.

Please, test it and comment to continue with import INFLOW.DAT (INFLOW.DAT is not read with that functionality yet).
(You need to create a new project from scratch to use the new field in the geaopackage)

**Comment by FLO-2DNoemi** [comment]
@FLO-2DJJ , I reviewed the latest plugin for the 2 Phase Flow component, please see below and comment:

- [ ] Mud and sediment transport dialog has to be reviewed, click OK and save data. 
- [ ] Read R line from INFLOW.DAT file with additional data for tailing tool elevation. Make sure we keep compatibility for other projects, for example R lines for water reservoirs, inflow hydrographs, etc.
- [ ] Model start running from QGIS.
- [ ] Review _Import GDS Files_ to read new data or files: INFLOW.DAT (R line), Tailings.DAT, SED.DAT and CON.DAT.
- [ ] Add Mudflow Data from Guidelines.

**Comment by FLO-2DKaren** [comment]
Testing now.

**Comment by FLO-2DNoemi** [comment]
@Juan Jose Rodriguez ***@***.***>

I downloaded the plugin and the new field for  tailings is not there.
Please review this.

**Comment by FLO-2DJJ** [comment]
Review the whole tailings functionality: Initial conditions Editor widget (Reservoirs), Grid Tools Widget (Sampling Tailings), Running Tailings Dam Breach Model

**Comment by FLO-2DKaren** [comment]
I've been using it for a while.  Import and export seems to work fine.  Data looks good.
